components-core/src/main/java/com/adyen/checkout/components/status/StatusRepository.kt

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2020 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by caiof on 18/8/2020.
+ */
+package com.adyen.checkout.components.status
+
+import android.os.Handler
+import android.os.Looper
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.adyen.checkout.components.status.api.StatusApi
+import com.adyen.checkout.components.status.api.StatusConnectionTask
+import com.adyen.checkout.components.status.api.StatusResponseUtils.isFinalResult
+import com.adyen.checkout.components.status.model.StatusResponse
+import com.adyen.checkout.core.api.Environment
+import com.adyen.checkout.core.exception.ApiCallException
+import com.adyen.checkout.core.exception.ComponentException
+import com.adyen.checkout.core.log.LogUtil.getTag
+import com.adyen.checkout.core.log.Logger
+import java.util.concurrent.TimeUnit
+
+class StatusRepository private constructor(environment: Environment) {
+
+    val handler = Handler(Looper.getMainLooper())
+    private val statusPollingRunnable: Runnable = object : Runnable {
+        override fun run() {
+            Logger.d(TAG, "mStatusPollingRunnable.run()")
+            statusApi.callStatus(clientKey!!, paymentData!!, mStatusCallback)
+            updatePollingDelay()
+            handler.postDelayed(this, pollingDelay)
+        }
+    }
+
+    val statusApi: StatusApi = StatusApi.getInstance(environment)
+    private val statusResponseLiveData = MutableLiveData<StatusResponse?>()
```

Comment:
The naming convention for private backing fields is usually the public field name prefixed with `_`, that way it's more clear which public and private fields belong together

Also, is nullability intended?

Reply:
Thx, changed the name. About the nullability unfortunately yes, check the comments on line 107.
This can obviously be improved now that we can change from LiveData to Flows.